src/note.cpp

```diff
@@ -459,6 +459,43 @@ TransPitch Note::GetTransPitch()
     return TransPitch(pname, this->GetChromaticAlteration(), this->GetOct());
 }
 
+void Note::UpdateFromTransPitch(TransPitch tp)
+{
+    data_PITCHNAME pname = static_cast<data_PITCHNAME>(tp.m_pname + PITCHNAME_c);
+    this->SetPname(pname);
+
+    Accid *accid = this->GetDrawingAccid();
+    if (accid->HasAccidGes()) {
+        data_ACCIDENTAL_GESTURAL gestural = ACCIDENTAL_GESTURAL_NONE;
+        switch (tp.m_accid) {
+            case -2: gestural = ACCIDENTAL_GESTURAL_ff; break;
+            case -1: gestural = ACCIDENTAL_GESTURAL_f; break;
+            case 0: gestural = ACCIDENTAL_GESTURAL_NONE; break;
+            case 1: gestural = ACCIDENTAL_GESTURAL_s; break;
+            case 2: gestural = ACCIDENTAL_GESTURAL_ss; break;
+            default: break; // TODO: Return an error here.
+        }
+        accid->SetAccidGes(gestural);
+    }
+    else {
```

Comment:
>  For @accid/@accid.ges, the output accid data must be placed into the same location from where it was extracted (this is important).

I think we should be changing both of these, because sometimes, both of them are given: 
```
<accid accid="s" accid.ges="s" enclose="paren" />
<accid accid="x" accid.ges="ss" />
<accid accid="f" accid.ges="f" />
```

Reply:
(1) and (2) have already been discussed, and that is correct.

(3) has two subcases: (3a) both are given and both are the same.  In this case, both should be changed to the same values (as appropriate for their data type).  The official view is that if both are the same, then only `@accid` should be given -- this is probably noted by the MEI Schematron validation. 
 (3b):  Both are present, but they are not the same.  There can be many reasons for this, some of which could be anticipated and some not.  I think a warning should be printed when this case occurs, and use the  `@accid.ges` value (one warning regardless of whether or note there are more than one occurrences of the situation).  The problem is that the `@accid` will not be in a modern convention, so how to change it will not be certain.  The main legitimate case is for 17th-century music (and earlier) where a sharp sign will be visually displayed for a note that would otherwise be flat, and the sharp raises the flat to a natural.  I think in this case the `@accid` should remain a sharp and no changed, but there may be other conventions that would cause problems.

A warning message should also be given (once) if a microtonal accidental is encountered (we can think about microtonal transposition after the basics for the standard system has been implemented).

for (4) there are also two subcases: (4a) the note contains an implied `@accid.ges="n"`.  In which case that value should be used as input to the transposition system.  If the transposed note output has a non-natural accidental, then insert an explicit `@accid.ges` entry for that note with the updated accidental.  If the accidental from the transposition remains natural, then leave `@accid.ges` as implicit. (4b) there is no visual accidental, but there should be a non-natural `@accid.ges`.  The consensus is that this is poorly-formed data, and in any case cannot be transposed properly until the `@accid.ges` parameters are filled in correctly (the input is garbage, and the output will be garbage).  

I think for both sub-cases of (4), a warning message should be printed (one for the entire process).  @rettinghaus  will complain about that because he likes implicit `@accid.ges="n"`, so any data that he produces would generate the warning.  I don't like implicit `@accid.ges="n"` because then it is difficult to distinguish between cases (4a) which is good, and (4b) which is very bad.  In any case, I think transposing `@accid.ges` is the proper thing to do in this case.  When the key signature is transposed along with the notes, there should be no migration of the accidental between `@accid` and `@accid.ges`.  In other words, there should not be new accidentals being displayed visually on notes where there was no visual accidental.  So, in this case, there was no visual accidental in the input, so there should be no visual accidental in the output.

Reply:
3b: I think detecting mismatching accid and accid.ges is out of scope for this pull request. Implementing "transpose both" will have the effect of making 17th century music (and any other different conventions) follow modern conventions.

4: Makes sense. Transposed music should **not** retain implicit `@accid.ges="n"`, but I don't think a warning is necessary (or if so, it should be in `GetChromaticAlteration`, since it would also affect MIDI generation).

(updated in 3168822c)

Reply:
That is good for (4).

For (3b), I would transpose as requested but print a warning that both `@accid` and `@accid.ges` are used at the same time and are different values.  In such cases `@accid.ges` should be used for the transposition system.  The result will be correct sounding data, but the visual notation may be incorrect.  In the future, transposition can be tested on cases where the two attributes are present and not equal to see how `@accid` should be adjusted based on the transposition of `@accid.ges`.  In 17th-century music, the accidentals are usually treated as relative signs ("sharp" meaning raise by a semitone, rather than the modern absolute meaning of make the chromatic inflection of the diatonic value a sharp), so leaving a `@accid` as sharp when `@accid` goes from natural to sharp is the best thing to do in such cases.  But it gets complicated with how naturals are used, so correct transposition of the notation of 16th century and earlier would need more study of how the accidental signs behave when transposed.